### PR TITLE
feat(helm): allow labels to be set on dagit and daemon

### LINKED ISSUE #6414: Helm Chart should support adding labels to Dagit
## Use Case
We have some mutating webhooks that are triggered off specific labels we'd like to apply to a dagit k8s deployment, but the helm chart currently has no support for manually provided labels.

## Ideas of Implementation
It looks like there's currently support for adding additional annotations which can be copypasta'ed to adding labels.  I'm happy to implement this and submit a PR if it helps unblock on your end as well.

---
#### Message from the maintainers:
Excited about this feature? Give it a :thumbsup:. We factor engagement into prioritization.

### PR DESCRIPTION
**Pull request by rexledesma**
## Summary
As the title. Resolves https://github.com/dagster-io/dagster/issues/6414.

## Test Plan
pytest

### CODE REVIEW

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* master
  * **PR #6428** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/6428" target="_blank">Graphite</a>  (this PR)

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

**Comment by rexledesma** [comment]
This should also be done for the code server deployments, but currently `$deployment.labels` controls the pod labels. It would be a breaking change to make `$deployment.podLabels` control the pod labels now, and `$deployment.labels` to control the deployment labels.. waiting for the 0.14.0 cutoff to make that change

File: helm/dagster/schema/schema/charts/dagster/subschema/daemon.py [hunk 1]
```diff
@@ -68,6 +68,8 @@ class Daemon(BaseModel):
     env: Dict[str, str]
     envConfigMaps: List[kubernetes.ConfigMapEnvSource]
     envSecrets: List[kubernetes.SecretEnvSource]
+    labels: Dict[str, str]
+    podLabels: Dict[str, str]
```
Comment (gibsondan) [inline helm/dagster/schema/schema/charts/dagster/subschema/daemon.py:72]: so we already have `labels` in the run launcher and they apply to the pod not the job - given that i think we should swap which one is the default? (so labels and deploymentLabels)?

**Review by gibsondan — DISMISSED** [review]
small naming thing for consistency, otherwise looks good

**Review by gibsondan — APPROVED** [review]

**Pull request merged**